src/Index/Index.php

```diff
@@ -15,14 +15,31 @@ class Index implements ReadableIndex, \Serializable
     use EmitterTrait;
 
     /**
-     * An associative array that maps fully qualified symbol names to Definitions
+     * An associative array that maps namespaces to
+     *     an associative array that maps fully qualified symbol names
+     *     to global Definitions, e.g. :
+     *     [
+     *         'Psr\Log\LoggerInterface' => [
+     *             'Psr\Log\LoggerInterface->log()' => $definition,
+     *             'Psr\Log\LoggerInterface->debug()' => $definition,
```

Comment:
What about using a full tree? e.g.
```php
[
  'Psr' => [
    '\Log' => [
      '\LoggerInterface' => [
        '->log()' => $definition
      ]
    ]
  ]
]

Reply:
What do you mean with "mainly"? Autocompletion for classes needs to get all classes in a namespace

Reply:
Even if it's not a full tree, this should duplicate the prefix on the second level. That's redundant data.

Reply:
Hello,
I started to change the index to use a full tree as you suggested.
I still have to use two arrays to avoid definition collitions : 
```php
[
    'Psr' => [
        '\Log' => [
            '\LoggerInterface' => $definition, // the non member definition
        ],
    ],
]
```
```php
[
    'Psr' => [
        '\Log' => [
            '\LoggerInterface' => [ // here it's an array, not a non-member definition
                '->log()' => $definition, // the member definition
            ],
        ],
    ],
]
```

I haven't pushed it yet as the tests aren't passing on my machine.

Reply:
That looks great! Instead of two arrays, you can just use the empty string as a key for the LoggerInterface itself